### PR TITLE
Adds homepage with user icon and logout option

### PR DESCRIPTION
**Pull request by balwant-jtc**
## Description
Adds homepage with user icon and logout option

https://github.com/jalantechnologies/boilerplate-mern/assets/78337443/63d180da-39be-4d84-bc53-c17331a4e6ca

### CODE REVIEW

File: src/apps/frontend/pages/header/header.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+      label: 'Logout',
+      onClickHandler: handleSignOut,
+    },
+  ];
+  return (
+    <HeaderNavigation>
+      <NavbarDesktopLeftSection />
+      <NavbarDesktopCenterSection />
+      <NavbarDesktopRightSection
```
Comment (jjalan) [inline src/apps/frontend/pages/header/header.tsx:32]: Why `Desktop`?

In a responsive web, I don't expect to see separate `web` or `mobile` components. The rendering / stacking of components can change but no specific components for web or mobile

File: src/apps/frontend/pages/header/header.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        profileMenuItems={profileMenuItems}
+        userAvatarUrl="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
```
Comment (jjalan) [inline src/apps/frontend/pages/header/header.tsx:34]: No No No.

This does not look nice.

Can we look into some icon lib in react?

**Review by jjalan — COMMENTED** [review]
I am left with a desire to see more polished header and footer design.

I am not saying color or jazzy interaction but how the login screen has elegance, that is missing from dashboard page.

File: src/apps/frontend/contexts/auth.provider.tsx [hunk 3]
```diff
@@ -11,9 +11,11 @@ import useAsync from './async.hook';
 
 type AuthContextType = {
   isLoginLoading: boolean;
+  isUserAuthenticated: () => boolean;
   loginError: AsyncError;
   loginResult: AccessToken;
   signIn: (username: string, password: string) => Promise<AccessToken>;
+  signOut: () => void;
 };
```
[the remark this answers is not in the document]
Comment (balwant-jtc) [reply src/apps/frontend/contexts/auth.provider.tsx]: Addressed

File: src/apps/frontend/pages/authentication/login/index.tsx [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
 import AuthenticationFormLayout from '../authentication-form-layout';
 import AuthenticationPageLayout from '../authentication-page-layout';
 
 import LoginForm from './login-form';
 
 export const Login: React.FC = () => {
+  const navigate = useNavigate();
   const onSuccess = () => {
-    toaster.positive(constant.LOGIN_SUCCESS_MSG);
+    navigate('/app');
```
[the remark this answers is not in the document]
Comment (balwant-jtc) [reply src/apps/frontend/pages/authentication/login/index.tsx]: Addressed

File: src/apps/frontend/pages/header/navbar-sections/navbar-desktop-right-section.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  profileMenuItems: ProfileMenuItem[];
+  userAvatarUrl: string;
+}
+
+const NavbarDesktopRightSection: React.FC<NavbarDesktopRightSectionProps> = ({
+  profileMenuItems,
+  userAvatarUrl,
+}) => (
+  <StyledNavigationList $align={ALIGN.right}>
+    <StyledNavigationItem style={{ padding: '0 40px' }}>
```
[the remark this answers is not in the document]
Comment (balwant-jtc) [reply src/apps/frontend/pages/header/navbar-sections/navbar-desktop-right-section.tsx]: Addressed

File: src/apps/frontend/pages/header/header.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  const profileMenuItems = [
+    {
+      label: 'Logout',
+      onClickHandler: handleSignOut,
+    },
+  ];
+  return (
+    <HeaderNavigation>
+      <NavbarDesktopLeftSection />
+      <NavbarDesktopCenterSection />
```
[the remark this answers is not in the document]
Comment (balwant-jtc) [reply src/apps/frontend/pages/header/header.tsx]: Addressed

File: src/apps/frontend/pages/header/header.tsx (hunk 1 above)
Comment (balwant-jtc) [reply src/apps/frontend/pages/header/header.tsx]: Addressed

File: src/apps/frontend/pages/header/header.tsx (hunk 2 above)
Comment (balwant-jtc) [reply src/apps/frontend/pages/header/header.tsx]: Addressed

File: src/apps/frontend/contexts/auth.provider.tsx [hunk 7]
```diff
@@ -33,19 +35,27 @@ const login = async (
   return result;
 };
 
+const logoutFn = (): void => localStorage.removeItem('access-token');
+
+const getAccessToken = (): AccessToken => JSON.parse(localStorage.getItem('access-token')) as AccessToken;
+
+const isUserAuthenticated = () => !!getAccessToken();
```
Comment (jjalan) [inline src/apps/frontend/contexts/auth.provider.tsx:42]: I would expect them to be part of `AuthProvider`.

Why are they outside?

**Review by jjalan — COMMENTED** [review]
Can you please update the video so I can see the latest experience?

File: src/apps/frontend/pages/header/header.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+interface HeaderNavbarProps {}
+
+const Header: React.FC<HeaderNavbarProps> = () => {
+  const navigate = useNavigate();
+  const { logout } = useAuthContext();
+
+  const handleSignOut = () => {
+    logout();
+    navigate('/login');
```
Comment (jjalan) [inline src/apps/frontend/pages/header/header.tsx:18]: Create a routes file instead of hardcoding the path

**Review by jjalan — APPROVED** [review]

File: src/apps/frontend/pages/header/header.tsx (hunk 8 above)
Comment (balwant-jtc) [reply src/apps/frontend/pages/header/header.tsx:18]: Addressed

**Comment by balwant-jtc** [comment]
> Can you please update the video so I can see the latest experience?

Done

File: src/apps/frontend/contexts/auth.provider.tsx [hunk 9]
```diff
@@ -33,19 +35,27 @@ const login = async (
...
+
 export const AuthProvider: React.FC<PropsWithChildren> = ({ children }) => {
   const {
     isLoading: isLoginLoading,
     error: loginError,
     result: loginResult,
-    asyncCallback: signIn,
-  } = useAsync(login);
+    asyncCallback: login,
+  } = useAsync(loginFn);
```
Comment (jjalan) [inline src/apps/frontend/contexts/auth.provider.tsx:50]: bookmark

**Pull request merged**